contracts/adapters/KycOnboarding.sol

```diff
@@ -0,0 +1,352 @@
+     * @param signature is message signature for verification
+     */
+    function _onboard(
+        DaoRegistry dao,
+        address kycedMember,
+        address tokenAddr,
+        bytes memory signature
+    ) internal reentrancyGuard(dao) {
+        require(!dao.isActiveMember(dao, kycedMember), "already member");
+
+        require(
+            dao.getNbMembers() <
+                dao.getConfiguration(_configKey(tokenAddr, MaxMembers)),
+            "the DAO is full"
+        );
+
+        _checkKycCoupon(dao, kycedMember, tokenAddr, signature);
+        OnboardingDetails memory details = _checkData(dao, tokenAddr);
+
+        BankExtension bank = BankExtension(
+            dao.getExtensionAddress(DaoHelper.BANK)
+        );
+        DaoHelper.potentialNewMember(kycedMember, dao, bank);
+        totalUnits[dao][tokenAddr] += details.unitsRequested;
+        address payable multisigAddress = payable(
+            dao.getAddressConfiguration(
+                _configKey(tokenAddr, FundTargetAddress)
+            )
+        );
+        if (multisigAddress == address(0x0)) {
+            if (tokenAddr == DaoHelper.ETH_TOKEN) {
+                bank.addToBalance{value: details.amount}(
+                    DaoHelper.GUILD,
+                    DaoHelper.ETH_TOKEN,
+                    details.amount
+                );
+            } else {
+                bank.addToBalance(DaoHelper.GUILD, tokenAddr, details.amount);
+                IERC20 erc20 = IERC20(tokenAddr);
+                erc20.safeTransferFrom(
+                    msg.sender,
+                    address(bank),
+                    details.amount
+                );
+            }
+        } else {
+            if (tokenAddr == DaoHelper.ETH_TOKEN) {
+                _weth.deposit{value: details.amount}();
+                _weth20.safeTransferFrom(
+                    address(this),
+                    multisigAddress,
+                    details.amount
+                );
+            } else {
+                IERC20 erc20 = IERC20(tokenAddr);
+                erc20.safeTransferFrom(
+                    msg.sender,
+                    multisigAddress,
+                    details.amount
+                );
+            }
+        }
+
+        bank.addToBalance(kycedMember, DaoHelper.UNITS, details.unitsRequested);
+
+        if (msg.value > details.amount) {
+            payable(msg.sender).sendValue(msg.value - details.amount);
+        }
+
+        emit Onboarded(dao, kycedMember, details.unitsRequested);
+    }
+
+    function _checkData(DaoRegistry dao, address tokenAddr)
+        internal
+        view
+        returns (OnboardingDetails memory details)
+    {
+        details.chunkSize = uint88(
+            dao.getConfiguration(_configKey(tokenAddr, ChunkSize))
+        );
+        require(details.chunkSize > 0, "config chunkSize missing");
+        details.numberOfChunks = uint88(msg.value / details.chunkSize);
```

Comment:
Is it the idea that you need to send eth value when you try to onboard with an ERC20? I mean, if it is an ERC20 the value will be 0, no?